lib/fcm.rb

```diff
@@ -150,70 +122,68 @@ def recover_notification_key(key_name, project_id)
     end
   end
 
-  def send_with_notification_key(notification_key, options = {})
-    body = { to: notification_key }.merge(options)
-    execute_notification(body)
-  end
-
-  def topic_subscription(topic, registration_id)
+  def topic_subscription(topic, registration_token)
     for_uri(INSTANCE_ID_API) do |connection|
-      response = connection.post("/iid/v1/#{registration_id}/rel/topics/#{topic}")
+      response = connection.post(
+        "/iid/v1/#{registration_token}/rel/topics/#{topic}"
+      )
       build_response(response)
     end
   end
 
-  def batch_topic_subscription(topic, registration_ids)
-    manage_topics_relationship(topic, registration_ids, "Add")
+  def topic_unsubscription(topic, registration_token)
+    batch_topic_unsubscription(topic, [registration_token])
   end
 
-  def batch_topic_unsubscription(topic, registration_ids)
-    manage_topics_relationship(topic, registration_ids, "Remove")
+  def batch_topic_subscription(topic, registration_tokens)
+    manage_topics_relationship(topic, registration_tokens, 'Add')
   end
 
-  def manage_topics_relationship(topic, registration_ids, action)
-    body = { to: "/topics/#{topic}", registration_tokens: registration_ids }
+  def batch_topic_unsubscription(topic, registration_tokens)
+    manage_topics_relationship(topic, registration_tokens, 'Remove')
+  end
+
+  def manage_topics_relationship(topic, registration_tokens, action)
+    body = { to: "/topics/#{topic}", registration_tokens: registration_tokens }
 
     for_uri(INSTANCE_ID_API) do |connection|
       response = connection.post("/iid/v1:batch#{action}", body.to_json)
       build_response(response)
     end
   end
 
-  def send_to_topic(topic, options = {})
-    if topic.gsub(TOPIC_REGEX, "").length == 0
-      send_with_notification_key("/topics/" + topic, options)
-    end
-  end
-
   def get_instance_id_info(iid_token, options = {})
     params = options
 
     for_uri(INSTANCE_ID_API) do |connection|
-      response = connection.get("/iid/info/" + iid_token, params)
+      response = connection.get("/iid/info/#{iid_token}", params)
       build_response(response)
     end
   end
 
-  def subscribe_instance_id_to_topic(iid_token, topic_name)
-    batch_subscribe_instance_ids_to_topic([iid_token], topic_name)
-  end
-
-  def unsubscribe_instance_id_from_topic(iid_token, topic_name)
-    batch_unsubscribe_instance_ids_from_topic([iid_token], topic_name)
-  end
-
-  def batch_subscribe_instance_ids_to_topic(instance_ids, topic_name)
-    manage_topics_relationship(topic_name, instance_ids, "Add")
-  end
-
-  def batch_unsubscribe_instance_ids_from_topic(instance_ids, topic_name)
-    manage_topics_relationship(topic_name, instance_ids, "Remove")
+  def send_to_topic(topic, options = {})
+    if topic.gsub(TOPIC_REGEX, '').zero?
```

Comment:
#gsub(pattern, replacement) returns String
(https://ruby-doc.org/core-3.1.1/String.html#method-i-gsub).
#zero? is undefined for String.
It will fell with 'NoMethodError: undefined method `zero?` for "":String'.

Reply:
Thank you for this, I have added back `.length` and also added test for this